### PR TITLE
don't fall off the end of the deltas into garbage

### PR DESCRIPTION
**Pull request by wjblanke**
./ProofOfSpace prove ffffff11239639109fd2ecdfff3f51457f9f805a5294fbee660a5f89c9e11683

### CODE REVIEW

**Review by mariano54 — APPROVED** [review]

**Review by mariano54 — CHANGES_REQUESTED** [review]
Actually this is still not working for me.
 ./ProofOfSpace prove fffffa2b647d4651c500076d7df4c6f352936cf293bd79c591a7b08e43d6adfb
 plot id: plot-k22-cc01d66ab532c2293e44a3a5f14325b92991ed4d5bf5a4c69efb9f766e0d0326.plot

**Comment by mariano54** [comment]
I'm working on a fix now.

**Review by hoffmang9 — APPROVED** [review]
+1

**Review by hoffmang9 — APPROVED** [review]
Seems like my approval didn't stick?

**Comment by mariano54** [comment]
Not sure why the test fail... But I tried chia plots check and it seemed good

**Review by mariano54 — APPROVED** [review]

File: src/prover_disk.hpp [hunk 1]
```diff
@@ -326,8 +326,10 @@ class DiskProver {
         std::vector<uint8_t> deltas =
             Encoding::ANSDecodeDeltas(bit_mask, encoded_size, kCheckpoint1Interval, kC3R);
         std::vector<uint64_t> p7_positions;
+        bool surpassed_f7 = false;
         for (uint8_t delta : deltas) {
             if (curr_f7 > f7) {
+                surpassed_f7 = true;
```
Comment (arvidn) [inline src/prover_disk.hpp:332]: perhaps I'm missing something, but it looks like we could just return `p2_positions` here, remove the `surpassed_f7` boolean and return an empty vector unconditionally at the end of the function.

File: src/prover_disk.hpp [hunk 2]
```diff
@@ -337,12 +339,19 @@ class DiskProver {
...
+            // we have an actual solution for f7.
             if ((int64_t)curr_p7_pos >= (int64_t)((c1_index + 1) * kCheckpoint1Interval) - 1 ||
-                curr_f7 >= (((uint64_t)1) << k)) {
-                return p7_positions;
+                curr_f7 >= (((uint64_t)1) << k) - 1) {
+                break;
             }
         }
+        if (!surpassed_f7) {
+            return std::vector<uint64_t>();
```
Comment (arvidn) [inline src/prover_disk.hpp:353]: I'm pretty sure this could even just be `return {};`

File: src/prover_disk.hpp (hunk 1 above)
Comment (mariano54) [reply src/prover_disk.hpp:332]: Yeah that is simpler..

**Pull request merged**